File: java/unidad_03/Problema_3_04/src/Problema_3_04.java
import java.util.InputMismatchException;
import java.util.Scanner;

/*
 * El programa, teniendo en cuenta ciertos criterios, calcula el aumento de sueldo
 * para un grupo de trabajadores. Imprime el nuevo sueldo del trabajador y la nómina
 * correspondiente.
 */

public class Problema_3_04 {

    public static float ingresarSueldo(int noEmpleado, Scanner scanner) {

        float sueldo;

        System.out.printf("Sueldo del trabajador %d: ", noEmpleado);
        sueldo = scanner.nextFloat();

        if (sueldo < -1) {

            System.out.println("Error: el sueldo debe ser mayor que cero.");
            return -1;
        }
        return sueldo;
    }

    public static void main(String[] args) {

        float sueldo;

        int noEmpleado = 1;

        float nuevoSueldo;

        float nomina = 0;

        try (Scanner scanner = new Scanner(System.in)) {

            sueldo = ingresarSueldo(noEmpleado, scanner);

            while (sueldo != -1) {

                if (sueldo < 1000) {

                    nuevoSueldo = sueldo * (float) 1.15;

                } else {

                    nuevoSueldo = sueldo * (float) 1.12;
                }
                nomina += nuevoSueldo;

                System.out.printf("Nuevo sueldo: %.2f %n %n", nuevoSueldo);
                noEmpleado++;

                sueldo = ingresarSueldo(noEmpleado, scanner);
            }
            System.out.printf("Nómina de la empresa: %.2f", nomina);

        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
    }
}

File: python/unidad_03/problema_3_04.py
'''
El programa, teniendo en cuenta ciertos criterios, calcula el aumento de sueldo
para un grupo de trabajadores. Imprime el nuevo sueldo del trabajador y la nómina
correspondiente.
'''

def ingresar_sueldo(no_empleado):

    sueldo = float(input(f"Sueldo del trabajador {no_empleado}: "))

    if sueldo < -1:

        print("Error: el sueldo debe ser mayor que cero.")
        return -1

    return sueldo

def main():

    no_empleado = 1

    nomina = 0

    try:

        sueldo = ingresar_sueldo(no_empleado)

        while sueldo != -1:

            if sueldo < 1000:

                nuevo_sueldo = sueldo * 1.15

            else:

                nuevo_sueldo = sueldo * 1.12
            
            nomina += nuevo_sueldo

            print(f"Nuevo sueldo: {nuevo_sueldo:.2f} \n")
            no_empleado += 1

            sueldo = ingresar_sueldo(no_empleado)

        print(f"Nómina de la empresa: {nomina:.2f}")

    except ValueError as e:

        print(e)
        return

if __name__ == '__main__':

    main()